src/Gui/DlgGeneralImp.cpp

```diff
@@ -104,6 +106,15 @@ DlgGeneralImp::DlgGeneralImp( QWidget* parent )
     else
         ui->RevertToSavedConfig->setEnabled(true);
     connect(ui->RevertToSavedConfig, &QPushButton::clicked, this, &DlgGeneralImp::revertToSavedConfig);
+
+    connect(ui->WBSLeftCorner, SIGNAL(toggled(bool)),
+        this, SLOT(emitReloadWorkbenchNeeded(bool)));
+    connect(ui->WBSRightCorner, SIGNAL(toggled(bool)),
+        this, SLOT(emitReloadWorkbenchNeeded(bool)));
+    connect(ui->WBSToolBar, SIGNAL(toggled(bool)),
+        this, SLOT(emitReloadWorkbenchNeeded(bool)));
+    connect(ui->WBSViewMenu, SIGNAL(toggled(bool)),
+        this, SLOT(emitReloadWorkbenchNeeded(bool)));

```

Comment:
No. The logic to take account into preference change shall be implemented in a ParameterObserver. Probably one already exists in `MainWindow` or `WorkbenchManager` -- have a look to other neighboring prefs --. I can further help if unclear.

Reply:
All of this and needReloadWb (see below comment) are because the changes were not taken into account when saving. It needed a reload of workbench (which reloaded the menubar/toolbars).
Do you know how to do it a clean way?

Reply:
As said, you shall get the parameter modifications through an observer.
A good example is probably the StatusBarObserver in MainWindow : https://github.com/FreeCAD/FreeCAD/blob/81143c17a17997a95a25dc98cb28fddf4d5272a8/src/Gui/MainWindow.cpp#L2056
All actions that are needed in case of parameter change shall be handled in the observer.